virttest/guest_agent.py

```diff
@@ -732,3 +732,90 @@ def fsthaw(self, check_status=True):
                     raise
             return ret
         return -1
+
+    def guest_file_operation(self, operation, **kwargs):
+        """
+        Basic guest file command function.
+
+        :param operation: open,write,read,flush,seek or flush.
+        :param kwargs: optional keyword arguments including but not limited
+               to below.
+        :return: The command's output.
+        """
+        cmd = "guest-file-%s" % operation
+        self.check_has_command(cmd)
+
+        if "con_encode" in kwargs:
+            args = {"buf-b64": kwargs["con_encode"]}
+            kwargs.pop("con_encode")
+            kwargs.update(args)
+
+        return self.cmd(cmd=cmd, args=kwargs)
+
+    def guest_file_open(self, path, mode="r"):
+        """
+        Open a guest file.
+
+        :param path: full path to the file in the guest to open.
+        :param mode: optional open mode, "r" is the default value.
+        :return: file handle.
+        """
+        return self.guest_file_operation("open", path=path, mode=mode)
+
+    def guest_file_close(self, handle):
+        """
+        Close an opened guest file.
+
+        :param handle: file handle returned by guest-file-open.
+        :return: no-error return.
+        """
+        return self.guest_file_operation("close", handle=handle)
+
+    def guest_file_write(self, handle, content):
+        """
+        Write to guest file.
+
+        :param handle: file handle returned by guest-file-open.
+        :param content: content to write.
+        :return: a dict with count and eof.
+        """
+        con_encode = base64.b64encode(content.encode()).decode()
+        return self.guest_file_operation("write", handle=handle,
+                                         con_encode=con_encode)
+
+    def guest_file_read(self, handle, count=None):
+        """
+        Read guest file.
+
+        :param handle: file handle returned by guest-file-open.
+        :param count: optional,maximum number of bytes to read(default is 4KB).
+        :return: a dict with base64-encoded string content,count and eof.
+        """
+        if count:
+            return self.guest_file_operation("read", handle=handle,
+                                             count=count)
+        else:
```

Comment:
don't need this `else`

Reply:
No,it's must.If count is not set, will read all content.

Reply:
updated,pls have a check.